src/apps/user/views.py

```diff
@@ -3,23 +3,63 @@
 from django.urls import reverse_lazy
 from django.views.generic import UpdateView, DetailView
 
-from src.apps.user.forms import CustomUserChangeForm
+from src.apps.user.forms import CustomUserChangeForm, ProfileForm
+from src.apps.user.models import UserProfile, UserAddress
 
 
-class ProfileDetailView(LoginRequiredMixin, DetailView):
-    template_name = "user/profile.html"
+class UserDetailView(LoginRequiredMixin, DetailView):
+    template_name = "user/user_detail.html"
     form_class = CustomUserChangeForm
-    success_url = reverse_lazy("user_profile")
+    success_url = reverse_lazy("user_detail")
 
     def get_object(self, queryset=None):
         return self.request.user
 
 
-class UserProfileUpdateView(LoginRequiredMixin, UpdateView):
+class UserUpdateView(LoginRequiredMixin, UpdateView):
     model = get_user_model()
     form_class = CustomUserChangeForm
-    template_name = "user/user_profile_edit.html"
-    success_url = reverse_lazy("user_profile")
+    template_name = "user/user_edit.html"
+    success_url = reverse_lazy("user_detail")
 
     def get_object(self, queryset=None):
         return self.request.user
+
+
+class ProfileUpdateView(LoginRequiredMixin, UpdateView):
+    model = UserProfile
+    form_class = ProfileForm
+    template_name = "profile/profile_edit.html"
+    success_url = reverse_lazy("profile_detail")
+
+    def get_object(self, queryset=None):
+        user = self.request.user
+        profile = UserProfile.objects.get(user=user)
+        if not profile.address:
+            address = UserAddress()
```

Comment:
Why do we need empty address? lets not create one unless user wants

Reply:
As far as i understand we are only creating it when user clicks update profile, so it is not created automatically(the same stuff is used as in profile creation before).

Reply:
well, 
lets not break single responsibility principle

get_object function should contain retrieve object logic only

Check SOLID metodology

Reply:
lgtm